sandbox/src/node_runner.rs

```diff
@@ -0,0 +1,132 @@
+use std::path::PathBuf;
+use std::process::{Child, Command};
+use std::sync::{Arc, RwLock};
+use std::time::Duration;
+
+use failure::Fail;
+use nix::sys::signal::{self, Signal};
+use nix::unistd::Pid;
+use wait_timeout::ChildExt;
+
+#[derive(Debug, Fail)]
+pub enum LightNodeRunnerError {
+    /// Already running error.
+    #[fail(display = "light node is already running")]
+    NodeAlreadyRunning,
+
+    /// Already running error.
+    #[fail(display = "light node is not running")]
+    NodeNotRunnig,
+
+    /// IO Error.
+    #[fail(display = "IO error during process creation")]
+    IOError { reason: std::io::Error },
+
+    /// Json argument parsing error.
+    #[fail(display = "Json argument parsing error")]
+    JsonParsingError,
+}
+
+impl From<std::io::Error> for LightNodeRunnerError {
+    fn from(err: std::io::Error) -> LightNodeRunnerError {
+        LightNodeRunnerError::IOError { reason: err }
+    }
+}
+
+/// Thread safe reference to a shared Runner
+pub type LightNodeRunnerRef = Arc<RwLock<LightNodeRunner>>;
+
+/// Struct that holds info about the running child process
+pub struct LightNodeRunner {
+    executable_path: PathBuf,
+    _name: String,
+    process: Option<Child>,
+    // TODO: anything else? Do we need a name? Maybe in the furture when launching multiple nodes with the sandbox
```

Comment:
@adonagy 
I think we should support multiple nodes now

Reply:
Let's merge this with just one node support for now. I created a task to add multinode support.